### PR TITLE
[release/1.3.1] [DEVOPS-398] Testnet genesis data

### PR DESCRIPTION
**Pull request by rvl**
This is based on top off #3585, which should be merged first.

## Description

- Adds more address discrimination to `configuration.yaml` and rearranges the way the testnet sections inherit their config.
- Updates the genesis data for testnet.

## Linked issue

- https://iohk.myjetbrains.com/youtrack/issue/DEVOPS-398
- https://iohk.myjetbrains.com/youtrack/issue/CO-354

## QA Steps

    nix-build -A connectScripts.testnet.wallet -o connect-testnet.sh
    ./connect-testnet.sh

## Screenshot

daedalus-testnet

### CODE REVIEW

**Comment by jmitchell** [comment]
> Please do not merge because `release/1.3.1` does not exist yet.

Until `release/1.3.1` has been cut from `develop` should this PR target `develop` instead of `release/1.3.0`? Then the resulting Daedalus installers for https://github.com/input-output-hk/daedalus/pull/1015 could more accurately reflect what `release/1.3.1` will eventually be.

**Comment by jmitchell** [comment]
> I understood that the hotfix release/1.3.1 will not be cut from develop, but rather from the v1.3.0 tag

Yes, my mistake.

**Comment by rvl** [comment]
Rebased on latest `release/1.3.0` and resolved merge conflicts in launch scripts/demo cluster.

File: lib/test/Test/Pos/ConfigurationSpec.hs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    rnm <- runIO (withConfigurations Nothing opts getRNM)
+    it "should be NMMustBeNothing" $
+        rnm `shouldBe` NMMustBeNothing
+
+getRNM :: NtpConfiguration -> ProtocolMagic -> IO RequiresNetworkMagic
+getRNM _ pm = pure (getRequiresNetworkMagic pm)
+
+spec :: Spec
+spec = mapM_ checkYamlSection [
+    "mainnet_full" -- mainnet core/relay nodes and exchange wallets
```
Comment (mhuesch) [inline lib/test/Test/Pos/ConfigurationSpec.hs]: Missing commas - I'll amend the commit and force-push.

Comment (disassembler) [reply lib/test/Test/Pos/ConfigurationSpec.hs]: thanks, I make that mistake all the time

**Comment by rvl** [comment]
On this branch I have tested:

```
nix-build -A connectScripts.testnet.wallet -o connect-testnet.sh && ./connect-testnet.sh
```

This sync successfully and proves that the validation errors I was seeing before is because I hadn't merged the right branch into my test branch.


```
nix-build -A connectScripts.mainnet.wallet -o connect-mainnet.sh && ./connect-mainnet.sh
```

This fails with:
```
[node:ERROR:ThreadId 5] [2018-09-19 02:22:01.02 UTC] logException: GenesisHashMismatch "AbstractHash 4832ba94116f22e49f0da1ff5e305ee7fe1a736fecf2fdbe6b8db31039148ae9" "AbstractHash 5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb"
cardano-node: GenesisHashMismatch "AbstractHash 4832ba94116f22e49f0da1ff5e305ee7fe1a736fecf2fdbe6b8db31039148ae9" "AbstractHash 5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb"
```
which means there is probably something wrong with the `mainnet_full` configuration in #3585 which needs to be fixed.

(If wondering how to debug this, `cat connect-mainnet.sh` then adjust the `cardano-node` command line to use your chosen binary and configuration file.)

**Comment by disassembler** [comment]
I'll try running the wallet on mainnet and staging after hydra builds everything. Unless you already have CO-354 in your branch (as of a few minutes ago) good chance that last commit resolves the issues you're having.

**Comment by rvl** [comment]
Thanks. Now that #3585 is merged, I have rebased on release/1.3.1 and also generated new genesis data from exactly this configuration.yaml (not one which was slightly different but effectively the same).

**Pull request merged**